### ISSUE TITLE
Make SimpleLinearGraphStage public?

### ISSUE THREAD

**Issue by He-Pin** [akka/akka#32006]
Currently when implement a GraphStage in Java require many boilerplate code.
![image](https://github.com/akka/akka/assets/501740/551df2ff-a5da-4b8e-9475-3c21220a461a)

I see there is a `SimpleLinearGraphStage` but which is `akka` private, I think it would be better to make it public and set a default GraphStageLogic for it, like the `Identity` stage.

**Comment by johanandren** [comment]
I don't think we should optimise for making custom graph stages minimal, also, all that the `SimpleLinearGraphStage` abstracts away is setting up the inlet/outlet and shape, so not a huge win.

Most simple linear cases should be covered by existing operators, especially now that we have `statefulMap`, instead of reaching for a custom graph stage.
